src/test/regress/sql/detect_conn_close.sql

```diff
@@ -0,0 +1,118 @@
+--
+-- PG15+ test as WL_SOCKET_CLOSED exposed for PG15+
+--
+SHOW server_version \gset
+SELECT substring(:'server_version', '\d+')::int >= 15 AS server_version_ge_15
+\gset
+\if :server_version_ge_15
+\else
+\q
+\endif
+
+CREATE SCHEMA socket_close;
+SET search_path TO socket_close;
+
+SET citus.shard_count TO 8;
+SET citus.shard_replication_factor TO 1;
```

Comment:
Missing citus.next_shard_id

In general I would recommend using `src/test/regress/bin/create_test.py` whenever you create a new test.

Reply:
> Missing citus.next_shard_id

Should we still set  shardids even when we don't rely on them? I only set when explicitly needed

Reply:
I guess it's fine not to set, but in my experience sometimes tests change later to rely on them.